sys/openbsd/init.go

```diff
@@ -113,6 +115,14 @@ func (arch *arch) neutralize(c *prog.Call, fixStructure bool) error {
 		if request.Val == arch.DIOCCLRSTATES || request.Val == arch.DIOCKILLSTATES {
 			request.Val = 0
 		}
+		// BIOCSETIF on tap leads to "tun: read failed"
+		if request.Val == arch.BIOCSETIF {
+			// Ideally this should also check Args[2] against "tap" as
+			// we only want to prevent the following from happening:
+			// ioctl$BIOCSETIF(-1, 0x8020426c, &(0x...)={'tap', 0x0})
```

Comment:
@dvyukov could you recommend the incantation to tighten this check?

Reply:
Are you interested in how exactly you could check for the `tap` string in the third argument or something more general, e.g. if there are other ways to restrict it?

Reply:
I just want this ioctl to not happen to `tap` interfaces because we rely on them working for packet injection.

Reply:
AFAIK there's currently to code that does something similar enough to this, but we could try to construct it from pieces.

1. Cast `c.Args[2]` to `*prog.PointerArg`. Let it be `ptrArg`.
2. Cast `ptrArg.Res` to `*prog.GroupArg`. Let it be `groupArg`.
3. Cast `groupArg.Inner[0]` to `*prog.DataArg`. Let it be `dataArg`.
4. Verify `dataArg.Data()` and patch it, if necessary. Like here:

https://github.com/google/syzkaller/blob/32fcf98fda0484949d799e870d7ac9945c695932/sys/linux/init.go#L241-L253

One this to watch out for are `ANY` types -- in that case, syzkaller has squahed the whole struct into one binary blob. We cannot really build a good tight sanity check in that case, so I think we can check for `ANY` and boldly patch it like you've already done it in this PR.

Here's an example of a check for ANY. You can do it for `c.Args[2]`.

https://github.com/google/syzkaller/blob/32fcf98fda0484949d799e870d7ac9945c695932/sys/linux/init_vusb.go#L61-L63

Reply:
Thanks for your guidance. 

I implemented 1-4 but I'm not sure where to get a `*prog.Gen` to call `.Target` on.

Also not sure if there's an easy way to test this without pushing it to prod :)

Reply:
> Also not sure if there's an easy way to test this without pushing it to prod :)

We do have quite a lot of tests for call sanitization in Linux: https://github.com/google/syzkaller/blob/master/sys/linux/init_test.go

>  I'm not sure where to get a *prog.Gen to call .Target on.

At the very least we could remember it in the `struct arch {}` -- the `prog.Target` pointer is passed to `InitTarget`.